addons/payment/static/src/js/payment_processing.js

```diff
@@ -94,7 +94,7 @@ odoo.define('payment.processing', function (require) {
                 return nbTx;
             }
             // if there's only one tx to manage
-            if(countTxInState(['tx_done', 'tx_error', 'tx_pending', 'tx_authorized']) === 1) {
+            if(countTxInState(['tx_done']) === 1 || countTxInState(['tx_done', 'tx_error', 'tx_pending', 'tx_authorized']) === 1) {
```

Comment:
Check if there is one transaction with the state `done` might seems to be a good fix, but I'd rather like to check if there is one transaction `done` post-processed. That way, we can't prevent a redirect if there is a problem at this stage. I've added a lot of comments to that suggestion, feel free to appropriate them but made sure to detail your fix :)

```suggestion
            /**
             * When the server sends the list of monitored transactions, it tries to post-process 
             * all the successful ones. If he succeeds or if the post-process has already been made, 
             * the transaction is removed from the list of monitored transactions and won't be 
             * included in the next response. We assume that successful and post-process 
             * transactions should always prevail on others, regardless of their number or state.
             */
            if (render_values['tx_done'].length === 1 &&
                render_values['tx_done'][0].is_post_processed) {
                    window.location = render_values['tx_done'][0].landing_route;
                    return;
            }

            // If there are multiple transactions monitored, display them all to the customer. If
            // there is only one transaction monitored, redirect directly the customer to the
            // landing route.
            if(countTxInState(['tx_done', 'tx_error', 'tx_pending', 'tx_authorized']) === 1) {
                // We don't want to redirect customers to the landing page when they have a pending
                // transaction. The successful transactions are dealt with before.
                var tx = render_values['tx_authorized'][0] || render_values['tx_error'][0];
```

Reply:
hi @chevalierv  thanks for your help it's done